packages/runtime-core/__tests__/apiWatch.spec.ts

```diff
@@ -857,4 +858,27 @@ describe('api: watch', () => {
 
     expect(instance!.effects![0].active).toBe(false)
   })
+
+  test('this.$watch should pass `this.proxy` to watch source as the first argument ', () => {
+    let args: any = null
+    let instance: any
+
+    const source = function() {
```

Comment:
use a `jest.fn()` instead and then you can `expect(source).toHaveBeenCalledWith(instance)`

Reply:
Done~